File: msgraph-mail/dotnet/Users/MailFolders/MailFolder.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users.MailFolders {
    public class MailFolder : Entity, IParsable<MailFolder> {
        public int ChildFolderCount { get; set; }
        public string DisplayName { get; set; }
        public string ParentFolderId { get; set; }
        public int TotalItemCount { get; set; }
        public int UnreadItemCount { get; set; }
        public List<MailFolder> ChildFolders { get; set; }
        public List<MessageRule> MessageRules { get; set; }
        public List<Message> Messages { get; set; }
        public List<MultiValueLegacyExtendedProperty> MultiValueExtendedProperties { get; set; }
        public List<SingleValueLegacyExtendedProperty> SingleValueExtendedProperties { get; set; }
        public new IDictionary<string, Action<MailFolder, IParseNode>> DeserializeFields => new Dictionary<string, Action<MailFolder, IParseNode>> {
            {
                "childFolderCount", (o,n) => { o.ChildFolderCount = n.GetIntValue(); }
            },
            {
                "displayName", (o,n) => { o.DisplayName = n.GetStringValue(); }
            },
            {
                "parentFolderId", (o,n) => { o.ParentFolderId = n.GetStringValue(); }
            },
            {
                "totalItemCount", (o,n) => { o.TotalItemCount = n.GetIntValue(); }
            },
            {
                "unreadItemCount", (o,n) => { o.UnreadItemCount = n.GetIntValue(); }
            },
            {
                "childFolders", (o,n) => { o.ChildFolders = n.GetCollectionOfObjectValues<MailFolder>().ToList(); }
            },
            {
                "messageRules", (o,n) => { o.MessageRules = n.GetCollectionOfObjectValues<MessageRule>().ToList(); }
            },
            {
                "messages", (o,n) => { o.Messages = n.GetCollectionOfObjectValues<Message>().ToList(); }
            },
            {
                "multiValueExtendedProperties", (o,n) => { o.MultiValueExtendedProperties = n.GetCollectionOfObjectValues<MultiValueLegacyExtendedProperty>().ToList(); }
            },
            {
                "singleValueExtendedProperties", (o,n) => { o.SingleValueExtendedProperties = n.GetCollectionOfObjectValues<SingleValueLegacyExtendedProperty>().ToList(); }
            },
        };
        public new void Serialize(ISerializationWriter writer) {
            base.Serialize(writer);
            writer.WriteIntValue("childFolderCount", ChildFolderCount);
            writer.WriteStringValue("displayName", DisplayName);
            writer.WriteStringValue("parentFolderId", ParentFolderId);
            writer.WriteIntValue("totalItemCount", TotalItemCount);
            writer.WriteIntValue("unreadItemCount", UnreadItemCount);
            writer.WriteCollectionOfObjectValues<MailFolder>("childFolders", ChildFolders);
            writer.WriteCollectionOfObjectValues<MessageRule>("messageRules", MessageRules);
            writer.WriteCollectionOfObjectValues<Message>("messages", Messages);
            writer.WriteCollectionOfObjectValues<MultiValueLegacyExtendedProperty>("multiValueExtendedProperties", MultiValueExtendedProperties);
            writer.WriteCollectionOfObjectValues<SingleValueLegacyExtendedProperty>("singleValueExtendedProperties", SingleValueExtendedProperties);
        }
    }
}

File: msgraph-mail/typescript/src/users/mailFolders/mailFolder.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';
import {Entity} from '../entity';
import {Message} from '../message';
import {MultiValueLegacyExtendedProperty} from '../multiValueLegacyExtendedProperty';
import {SingleValueLegacyExtendedProperty} from '../singleValueLegacyExtendedProperty';
import {MessageRule} from './messageRule';

export class MailFolder extends Entity implements Parsable<MailFolder> {
    public childFolderCount?: number | undefined;
    public displayName?: string | undefined;
    public parentFolderId?: string | undefined;
    public totalItemCount?: number | undefined;
    public unreadItemCount?: number | undefined;
    public childFolders?: MailFolder[] | undefined;
    public messageRules?: MessageRule[] | undefined;
    public messages?: Message[] | undefined;
    public multiValueExtendedProperties?: MultiValueLegacyExtendedProperty[] | undefined;
    public singleValueExtendedProperties?: SingleValueLegacyExtendedProperty[] | undefined;
    public serialize (writer: SerializationWriter) : void {
        super.serialize(writer);
        writer.writeNumberValue("childFolderCount", this.childFolderCount);
        writer.writeStringValue("displayName", this.displayName);
        writer.writeStringValue("parentFolderId", this.parentFolderId);
        writer.writeNumberValue("totalItemCount", this.totalItemCount);
        writer.writeNumberValue("unreadItemCount", this.unreadItemCount);
        writer.writeCollectionOfObjectValues<MailFolder>("childFolders", this.childFolders);
        writer.writeCollectionOfObjectValues<MessageRule>("messageRules", this.messageRules);
        writer.writeCollectionOfObjectValues<Message>("messages", this.messages);
        writer.writeCollectionOfObjectValues<MultiValueLegacyExtendedProperty>("multiValueExtendedProperties", this.multiValueExtendedProperties);
        writer.writeCollectionOfObjectValues<SingleValueLegacyExtendedProperty>("singleValueExtendedProperties", this.singleValueExtendedProperties);
    };
    public deserializeFields () : Map<string, (item: MailFolder, node: ParseNode) => void> {
        return new Map<string, (item: MailFolder, node: ParseNode) => void>([...super.deserializeFields(),
            ["childFolderCount", (o, n) => { o.childFolderCount = n.getNumberValue(); }],
            ["displayName", (o, n) => { o.displayName = n.getStringValue(); }],
            ["parentFolderId", (o, n) => { o.parentFolderId = n.getStringValue(); }],
            ["totalItemCount", (o, n) => { o.totalItemCount = n.getNumberValue(); }],
            ["unreadItemCount", (o, n) => { o.unreadItemCount = n.getNumberValue(); }],
            ["childFolders", (o, n) => { o.childFolders = n.getCollectionOfObjectValues<MailFolder>(MailFolder); }],
            ["messageRules", (o, n) => { o.messageRules = n.getCollectionOfObjectValues<MessageRule>(MessageRule); }],
            ["messages", (o, n) => { o.messages = n.getCollectionOfObjectValues<Message>(Message); }],
            ["multiValueExtendedProperties", (o, n) => { o.multiValueExtendedProperties = n.getCollectionOfObjectValues<MultiValueLegacyExtendedProperty>(MultiValueLegacyExtendedProperty); }],
            ["singleValueExtendedProperties", (o, n) => { o.singleValueExtendedProperties = n.getCollectionOfObjectValues<SingleValueLegacyExtendedProperty>(SingleValueLegacyExtendedProperty); }],
        ]);
    };
}
